src/core/border_router/router_advertisement.cpp

```diff
@@ -187,15 +187,27 @@ bool RouteInfoOption::IsValid(void) const
            (pref == OT_ROUTE_PREFERENCE_LOW || pref == OT_ROUTE_PREFERENCE_MED || pref == OT_ROUTE_PREFERENCE_HIGH);
 }
 
-RouterAdvMessage::RouterAdvMessage(void)
-    : mReachableTime(0)
-    , mRetransTimer(0)
+void RouterAdvMessage::SetToDefault(void)
 {
-    OT_UNUSED_VARIABLE(mReachableTime);
-    OT_UNUSED_VARIABLE(mRetransTimer);
-
     mHeader.Clear();
     mHeader.SetType(Ip6::Icmp::Header::kTypeRouterAdvert);
+    mReachableTime = 0;
+    mRetransTimer  = 0;
+}
+
+const RouterAdvMessage &RouterAdvMessage::operator=(const RouterAdvMessage &aOther)
+{
+    uint16_t checksum = mHeader.GetChecksum();
+
+    mHeader = aOther.mHeader;
+
+    // There is no necessary to learn the checksum.
+    mHeader.SetChecksum(checksum);
```

Comment:
Is there value to saving and restoring the checksum?

Reply:
We are currently using zero checksum value to let the platform fill the ICMPv6 checksum. I updated to set zero directly in the latest push.

Reply:
Also updated `platform/infra_if.h` to make this clearer.